app/models/miq_provision_workflow.rb

```diff
@@ -4,13 +4,15 @@ def self.base_model
   end
 
   def self.class_for_platform(platform)
-    "MiqProvision#{platform.classify}Workflow".constantize
+    "MiqProvision#{platform.classify}Workflow".safe_constantize ||
+      "ManageIQ::Providers::#{platform.classify}::CloudManager::ProvisionWorkflow".safe_constantize ||
+      "ManageIQ::Providers::#{platform.classify}::InfraManager::ProvisionWorkflow".constantize
   end
 
   def self.class_for_source(source_or_id)
-    source = source_or_id.kind_of?(ActiveRecord) ? source_or_id : VmOrTemplate.find_by_id(source_or_id)
+    source = VmOrTemplate.find_by_id(source_or_id)
     return nil if source.nil?
-    class_for_platform(source.class.model_suffix)
+    source.ext_management_system.class.provision_workflow_class
```

Comment:
Hmmm... we only need the EMS's class, so we might be able to arrange to go `source.class.ems_class` instead of `source.ems.class`. So yeah, we can plan to fix that when we hit it.